sdk/storage/azure-storage-blob/azure/storage/blob/aio/_download_async.py

```diff
@@ -559,11 +577,11 @@ async def readall(self) -> T:
         :rtype: T
         """
         stream = BytesIO()
-        await self.readinto(stream)
+        await self.readinto(stream)  # type: ignore [arg-type]
```

Comment:
Does `BytesIO` really not adhere to `IO[bytes]`? We should check if this error still exists. (Probably same for sync)